app/excel-parser/read-report.ts

```diff
@@ -0,0 +1,174 @@
+import * as XLSX from "xlsx";
+//const reportPath: string = "./10_Październik_2022.xlsx";
+const shiftNames: Array<string> = [
+  "04.10.2022 zm.A",
+  "04.10.2022 zm.B",
+  "04.10.2022 zm.C",
+];
+
+export class Part {
+  name: string;
+  ArticleNo: string;
+  tool: string;
+  machine: string;
+  totalPartsProduced: number;
+  scrapNo: number;
+  id: string;
+
+  constructor(
+    nameFromReport: string,
+    partNrFromReport: string,
+    toolFromReport: string,
+    machineFromReport: string,
+    productionFromReport: number,
+    scrapFromReport: number
+  ) {
+    this.name = nameFromReport;
+    this.ArticleNo = partNrFromReport;
+    this.tool = toolFromReport;
+    this.machine = machineFromReport;
+    this.totalPartsProduced = productionFromReport;
+    this.scrapNo = scrapFromReport;
+    this.id = this.ArticleNo + ";" + this.tool;
+  }
+}
+
+export function readExcel(reportPath: string) {
+  let woorkbookReport: XLSX.WorkBook = XLSX.readFile(reportPath);
+
+  let partList: Array<Part>;
+  let partDict: { [key: string]: Part } = {};
+
+  shiftNames.forEach(shiftNumber => {
+    console.log(shiftNumber)
+    let sheetShift: XLSX.Sheet = woorkbookReport.Sheets[shiftNumber];
+    partList = parseShift(sheetShift);
+
+    for (const part of partList) {
+      if (!(part.id in partDict)) {
```

Comment:
Don't use 'in' to check if property exists on a given object.
Use 'hasOwnProperty'

Detailed explanation:
https://masteringjs.io/tutorials/fundamentals/hasownproperty

Also just consider using Map for that purpose:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map